### PR TITLE
feat: no longer ship kubectl, instead alias k0s, use /var/lib/embedded-cluster

### PR DESCRIPTION
**Pull request by laverya**
This stops shipping kubectl as an independent binary, and instead creates it in `/var/lib/embedded-cluster/bin` as a symlink to `/usr/local/bin/k0s`. It also finally removes the config dir, as nothing uses it anymore.

Logs have been moved to `/var/lib/embedded-cluster/logs`.

### CODE REVIEW

**Comment by laverya** [comment]
I am not particularly happy with not being able to use the alias in the scripts, but I'm glad it works calling k0s directly

**Comment by laverya** [comment]
Should our installer run the kubectl `ln` instead?

That sounds like it might be a fantastic solution

**Comment by ricardomaraschini** [comment]
Autocomplete seems to have stopped to work with this.

**Comment by ricardomaraschini** [comment]
I decided to give this a try and this is what I saw:

```
root@ec:~# ec shell

    __4___
 _  \ \ \ \   Welcome to ec debug shell.
<'\ /_/_/_/   This terminal is now configured to access your cluster.
 ((____!___/) Type 'exit' (or CTRL+d) to exit.
  \0\0\0\0\/  Happy hacking.
 ~~~~~~~~~~~
 $(basename ${SHELL}))
root@ec:~/.config/ec/bin# alias kubectl="/usr/local/bin/k0s kubectl"export PATH="$PATH:/root/.config/ec/bin"
root@ec:~/.config/ec/bin# source <(kubectl completion $(basename ${SHELL}))
Error: unknown command "kubectlexport" for "k0s"
Run 'k0s --help' for usage.
root@ec:~/.config/ec/bin# source /etc/bash_completion
root@ec:~/.config/ec/bin#
```

This happens when I try to list the pods

```
root@ec:~/.config/ec/bin# kubectl get pods -A
Error: unknown command "kubectlexport" for "k0s"
Run 'k0s --help' for usage.
root@ec:~/.config/ec/bin#
```

**Comment by ricardomaraschini** [comment]
Interesting is that it seems like I could not even install:

```
root@ec:~# ec install --no-prompt
✔  Running host preflights on node
Admin Console password set to: password
✔  Node installation finished
✔  Embedded Cluster Operator is ready!
✔  Waiting for Admin Console to deploy: 0/3 ready
error waiting for admin console: error checking status of kotsadm: client rate limiter Wait returned an error: context canceled
root@ec:~#
```

It failed to wait for the Admin Console but it seems to be online:

```
root@ec:~# k0s kubectl get pods -A
NAMESPACE          NAME                                           READY   STATUS    RESTARTS       AGE
embedded-cluster   embedded-cluster-operator-5ddb94bf46-8wqps     2/2     Running   0              6m56s
kotsadm            kotsadm-54898d9-9qg6n                          1/1     Running   0              6m22s
kotsadm            kotsadm-minio-0                                1/1     Running   0              6m22s
kotsadm            kotsadm-rqlite-0                               1/1     Running   0              6m22s
kotsadm            kurl-proxy-kotsadm-64d646b5bc-qrmpm            1/1     Running   1 (6m4s ago)   6m22s
kube-system        calico-kube-controllers-84c6cd5b85-bjg78       1/1     Running   0              7m48s
kube-system        calico-node-vkvpp                              1/1     Running   0              7m42s
kube-system        coredns-6cd46fb86c-gtlwm                       1/1     Running   0              7m48s
kube-system        kube-proxy-tmwhl                               1/1     Running   0              7m42s
kube-system        metrics-server-7556957bb7-q5h84                1/1     Running   0              7m48s
openebs            openebs-localpv-provisioner-56d6489bbc-jfzrz   1/1     Running   0              7m48s
root@ec:~#
```

**Comment by laverya** [comment]
I'll figure out what's going on

**Review by chris-sanders — CHANGES_REQUESTED** [review]
See conversation: https://app.shortcut.com/replicated/story/94401/remove-all-unnecessary-files-created-on-worker-node-hosts-during-installation#activity-97013

We don't want to make files available on the host there should be no symlinking to `/usr/local`. If k0s is already installing in `/usr/local/bin` I can live with that because any other versions of k0s on the machine would be a problem and we need to stop anyway.

For the rest of our files though like 'kubectl' linking to `/usr/loca/bin` stands to conflict with existing kubectl installed on the machine. Instead we only need to set these values in our shell not machine wide.

**Comment by laverya** [comment]
@chris-sanders would you be OK with only creating the symlink if there was not already a kubectl?

**Comment by laverya** [comment]
Actually I'll make the symlink under `/var/lib/embedded-cluster/bins`

**Review by chris-sanders — APPROVED** [review]
The current work has incorporated my previous recommendations. I know it's still in flight and I'm approving it now so that my absence later in the day doesn't block the new work.

Someone should look at the new work when it's done :)

File: pkg/addons/adminconsole/adminconsole.go [hunk 1]
```diff
@@ -218,6 +217,7 @@ func (a *AdminConsole) GenerateHelmConfig(onlyDefaults bool) ([]v1beta1.Chart, [
 		if err := a.addKotsApplicationToHelmValues(); err != nil {
 			return nil, nil, fmt.Errorf("unable to add kots app to helm values: %w", err)
 		}
+		helmValues["embeddedClusterID"] = metrics.ClusterID().String()
```
Comment (sgalsaleh) [inline pkg/addons/adminconsole/adminconsole.go:220]: hmm, does this mean that there will be no `embeddedClusterID` when only defaults are used? TBH i'm not sure what `onlyDefaults` indicates.

**Review by sgalsaleh — DISMISSED** [review]

File: pkg/addons/adminconsole/adminconsole.go (hunk 1 above)
Comment (laverya) [reply pkg/addons/adminconsole/adminconsole.go:220]: onlyDefaults is what we pass in when running the `version metadata` command - it's what generates the metadata file that is associated with a embedded cluster version

it's shared, and thus really should not have an ID

**Review by sgalsaleh — APPROVED** [review]

**Comment by laverya** [comment]
I checked, and the shell was working!

**Pull request merged**